Makefile

```diff
@@ -60,7 +60,13 @@ edoc:
 		rm $${file}.bak ; \
 	done
 
-publish: edoc
+# Upgrade all dependencies and update lock files.
+# rebar3 requires the --all flag from v3.18.0
+upgrade:
+	@$(REBAR) upgrade --all || $(REBAR) upgrade
+	@mix deps.update --all
+
+publish: upgrade edoc
```

Comment:
After upgrading the deps, don't we need to build and test? Therefore, I'm not sure the upgrade should be part of publish.

Reply:
ah ok, this is here to make sure that we already use the latest dependencies.
The idea is that if the upgrade step will find anything new to upgrade it will generate new lock files, which this will block the publish in the `git status` step.
But if there was an upgrade during publish there might be a need to make sure to remember to run tests..

Reply:
Yeah, I think we should be able to publish without upgrading. Less dependencies between operations.

Reply:
One problem I noticed was that while `rebar3` supports notations like "~> 1.5" `rebar2` does not.
rebar2 supports regex-like notations like "1.5.*" 
We've had users before with issues regarding the use of unicode in rebar.config, which rebar2 doesn't support.
So, we might run into this problem in the future when/if changing to version-ranges like we have in mix.

Reply:
Good point.